File: services/administration-service/src/main/java/sgptt/adminsvc/entity/Activity.java
package sgptt.adminsvc.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import sgptt.adminsvc.model.ActivityName;

import java.util.Date;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Activity {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    @Column(name = "activity_id")
    private Long activityId;

    @Temporal(TemporalType.DATE)
    private Date openDate;

    @Temporal(TemporalType.DATE)
    private Date closeDate;

    @Enumerated(EnumType.ORDINAL)
    private ActivityName activity;
}

File: services/protocols-service/src/main/kotlin/com/sgptt/protocolsservice/repository/entity/Activity.kt
package com.sgptt.protocolsservice.repository.entity

import com.sgptt.protocolsservice.model.ActivityName
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Temporal
import jakarta.persistence.TemporalType
import java.util.*

@Entity
data class Activity(
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "activity_id")
	val activityId: Long,
	
	@Temporal(TemporalType.DATE)
	val openDate: Date,
	
	@Temporal(TemporalType.DATE)
	val closeDate: Date,
	
	@Enumerated(EnumType.ORDINAL)
	val activity: ActivityName,
)
